src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -198,17 +198,21 @@ const fetchDataSources = async (client: SavedObjectsClientContract): Promise<Dat
   });
   const dataSources: DataStructure[] = [DEFAULT_DATA.STRUCTURES.LOCAL_DATASOURCE];
   return dataSources.concat(
-    resp.savedObjects.map((savedObject) => ({
-      id: savedObject.id,
-      title: savedObject.attributes.title,
-      type: 'DATA_SOURCE',
-      meta: {
-        query: {
-          id: savedObject.id,
-        },
-        type: DATA_STRUCTURE_META_TYPES.CUSTOM,
-      } as DataStructureCustomMeta,
-    }))
+    resp.savedObjects
+      .filter(
+        (savedObject) => savedObject.attributes.dataSourceEngineType !== 'OpenSearch Serverless'
```

Comment:
Is attributes guaranteed to be defined?

Reply:
Attributes is guaranteed to be defined, but not sure about `dataSourceEngineType`. This should be some of the `data_source` plugin work using the saved objects client. I'm not super familiar, but it looks like it should also be defined for data sources [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/workspace/public/utils.ts#L249).
```
.then((response) => {
      const objects = response?.savedObjects;
      if (objects) {
        return objects.map((source) => {
          const id = source.id;
          // Data connection doesn't have title for now, would use connectionId instead to display.
          const title = source.get('title') ?? source.get('connectionId') ?? '';
          const workspaces = source.workspaces ?? [];
          const auth = source.get('auth');
          const description = source.get('description');
          const dataSourceEngineType = source.get('dataSourceEngineType');
          const type = source.type;
```